### PR TITLE
US105979 - add browser consistent version of offsetParent

### PR DESCRIPTION
**Pull request by AlexBedley**
Continuation of https://github.com/BrightspaceUI/tooltip/pull/45

Putting `offsetParent` in the correct place. I also changed it to use `getComposedParent`.

I started rewriting the tests to match the style of the other d2l-dom tests but was finding it not worth it (made the file messy and hard to read). So I just imported my previous written test file as a standalone and left the current tests as-is.

### CODE REVIEW

**Comment by AlexBedley** [comment]
Once this gets reviewed and merged, how do I release a new version of it to get it back to d2l-tooltip?

File: d2l-dom.js
Comment (dlockhart) [inline d2l-dom.js:85]: A bunch of things in here are going to [force layout/reflow](https://gist.github.com/paulirish/5d52fb081b3570c81e3a). @dbatiste, do you think we should make it async and use fastdom?

Comment (dlockhart) [reply d2l-dom.js:85]: Alternatively I guess whoever calls this could know that it's an alternate for `elem.offsetParent` which itself forces layout so they should use fastdom themselves.

Comment (dbatiste) [reply d2l-dom.js:85]: Yeah think that would be good to leave it to the consumer to use fast-dom since they might want to batch it with other measurements.  I think they could still batch it if we made this async though, but the consumers code might be a little more awkward.

We do use fast-dom a bunch throughout these helpers though - namely the vanilla expand/collapse, arrowkeys-behavior, and visible-on-ancestor - they are all doing higher level stuff though.

**Review by dlockhart — APPROVED** [review]
@AlexBedley once this merges, it'll release automatically as a minor version.

**Pull request merged**
